### PR TITLE
Remove response channel when it is not consumed anymore

### PR DESCRIPTION
**Pull request by b123400**
connect https://github.com/SkygearIO/skygear-server/issues/347

### CODE REVIEW

**Comment by oursky-travis** [comment]
@b123400, thanks for your PR! By analyzing the history of the files in this pull request, we identified @cheungpat to be a potential reviewer.

File: pkg/server/plugin/zmq/broker.go [hunk 1]
```diff
@@ -286,6 +286,15 @@ func (lb *Broker) setTimeout(address string) {
 	lb.timeout <- address
 }
 
+func (lb *Broker) cancelRPC(requestChan chan []byte) {
+	for k, v := range lb.addressChan {
```
Comment (cheungpat) [inline pkg/server/plugin/zmq/broker.go]: Only one coroutine should be modifying `lb.addressChan`.

File: pkg/server/plugin/zmq/rpc.go [hunk 2]
```diff
@@ -159,15 +159,17 @@ func (p *zmqTransport) ipc(req *pluginrequest.Request) (out []byte, err error) {
...
-	case msg := <-<-reqChan:
+	case msg := <- respChan:
 		// Broker will sent back a null byte if time out
 		if bytes.Equal(msg, []byte{0}) {
 			err = fmt.Errorf("Plugin time out")
 		} else {
 			out = msg
 		}
 	case <-req.Context.Done():
+		p.broker.cancelRPC(respChan)
```
Comment (cheungpat) [inline pkg/server/plugin/zmq/rpc.go]: This doesn’t seem to be robust because the context can be done and the request can finish at the same time.

**Comment by b123400** [comment]
Right, I think I will choose the prior

> Hold the guarantee that the caller will always consume from the channel. This requires creating a goroutine that always consume from the channel regardless of whether context is done.

**Comment by b123400** [comment]
Updated

**Pull request merged**